### ISSUE TITLE
prefix.join is not a function

### ISSUE THREAD

**Issue by markheron91** [mgechev/codelyzer#172]
Hi there,

Previous Working Versions: 

 - Codelyzer: 1.0.0-beta.4
 - TsLint: 3.15.1

Upgraded to the Following Versions:

 - Codelyzer: 2.0.0-beta.1
 - TsLint: 4.0.2

When upgrading from the previous to the new versions shown above, I can see the following error reported by the tslint extension in my vscode.

vscode-tslint: 'prefix.join is not a function' while validating: d:\Dev\Onevolve-BPO\src\app\api\application\application-api.service.ts stacktrace: TypeError: prefix.join is not a function
    at Rule.SelectorRule.setPrefixArguments (d:\Dev\Onevolve-BPO\node_modules\codelyzer\selectorNameBase.js:70:69)
    at Rule.SelectorRule [as constructor] (d:\Dev\Onevolve-BPO\node_modules\codelyzer\selectorNameBase.js:21:15)
    at new Rule (d:\Dev\Onevolve-BPO\node_modules\codelyzer\directiveSelectorRule.js:11:28)
    at Object.loadRules (d:\Dev\Onevolve-BPO\node_modules\tslint\lib\ruleLoader.js:46:32)
    at Linter.getEnabledRules (d:\Dev\Onevolve-BPO\node_modules\tslint\lib\linter.js:163:44)
    at Linter.lint (d:\Dev\Onevolve-BPO\node_modules\tslint\lib\linter.js:78:33)
    at doValidate (C:\Users\Mark\.vscode\extensions\eg2.tslint-0.6.4\server\server.js:291:20)
    at validateTextDocument (C:\Users\Mark\.vscode\extensions\eg2.tslint-0.6.4\server\server.js:210:27)
    at Delayer.d.trigger [as task] (C:\Users\Mark\.vscode\extensions\eg2.tslint-0.6.4\server\server.js:363:9)
    at completionPromise.Promise.then (C:\Users\Mark\.vscode\extensions\eg2.tslint-0.6.4\server\delayer.js:19:35)

The only things changed between those 2 versions is these parts as shown in the changelog:

(directive-selector-name & directive-selector-type) -> directive-selector
(component-selector-name & component-selector-type) -> component-selector

**Comment by markheron91** [comment]
Ok, think I've worked out what the problem might be, and if I'm right then it is more a problem with the changelog documentation. The changelog shows the directive-selector-name and directive-selector-type have been merged into one rule (directive-selector), but from looking at the code it looks like the directive-selector-prefix rule has also been merged to this rule aswell.

This was causing me to basically take the two arrays in the rules [true, 'attribute'] and [true, 'camelCase'] and merging them into one giving [true, 'attribute', true, 'camelCase']. The second true was causing the error in the code because join cannot be called on it.

Really what it should be changed to is [true, 'attribute', 'app', 'camelCase'] in my situation, merging in the prefix rule also. The changelog has the "sg" in the third element of the array but it is not clear when reading it that the "sg" is actually supposed to come from the directive-selector-prefix rule that is also removed.

I think the changelog should be updated to clarify the removal of these rules and how to properly combine them.
